nncf/common/logging/track_progress.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Callable, Iterable, List, Optional, Sequence, Union
+
+from rich.console import Console
+from rich.progress import BarColumn
+from rich.progress import Progress
+from rich.progress import ProgressColumn
+from rich.progress import ProgressType
+from rich.progress import Task
+from rich.progress import TaskProgressColumn
+from rich.progress import TextColumn
+from rich.progress import TimeRemainingColumn
+from rich.style import StyleType
+from rich.text import Text
+
+
+class IterationsColumn(ProgressColumn):
+    def render(self, task: Task) -> Text:
+        if task.total is None:
+            return Text("")
+        text = f"{int(task.completed)}/{int(task.total)}"
+        if task.finished:
+            return Text(text, style="progress.elapsed")
+        return Text(text, style="progress.remaining")
+
+
+def track(
+    sequence: Union[Sequence[ProgressType], Iterable[ProgressType]],
+    description: str = "Working...",
+    total: Optional[float] = None,
+    auto_refresh: bool = True,
+    console: Optional[Console] = None,
+    transient: bool = False,
+    get_time: Optional[Callable[[], float]] = None,
+    refresh_per_second: float = 10,
+    style: StyleType = "bar.back",
+    complete_style: StyleType = "bar.complete",
+    finished_style: StyleType = "bar.finished",
+    pulse_style: StyleType = "bar.pulse",
+    update_period: float = 0.1,
+    disable: bool = False,
+    show_speed: bool = True,
+) -> Iterable[ProgressType]:
+    """
+    Track progress by iterating over a sequence.
+
+    This function is very similar to rich.progress.track(), but with addition of an iteration counter.
+
+    Args:
+        sequence (Iterable[ProgressType]): A sequence (must support "len") you wish to iterate over.
```

Comment:
Docstring does not conform to our style guide.

Reply:
Indeed, I propose just to mention the `rich.progress.track()` as the source of the information for the params and not rewrite the docstring.

Reply:
I decided to keep reformatted docstring. This way it will be easier to use it.